server/src/main/java/org/opensearch/index/mapper/DocumentParser.java

```diff
@@ -427,6 +427,8 @@ private static void innerParseObject(
     ) throws IOException {
         assert token == XContentParser.Token.FIELD_NAME || token == XContentParser.Token.END_OBJECT;
         String[] paths = null;
+        context.incrementFieldCurrentDepth();
+        context.checkFieldDepthLimit();
```

Comment:
@nkumar04 question here, so we focus on object nesting but what about arrays nesting, shouldn't we check this part as well?

```
{
	"nested": [
		[
			[
			  [
				 // go on ...
			  ]
			]
		],
		[
			[
			  [
				 // go on ...
			  ]
			]
		]
	]
}
```


Test case to reproduce (fails):

```
    public void testDocumentContainsDeepNestedFieldParsingShouldFail() throws Exception {
        DocumentMapper mapper = createDocumentMapper(fieldMapping(b -> {
            b.field("type", "nested");
        }));

        long depth_limit = MapperService.INDEX_MAPPING_DEPTH_LIMIT_SETTING.getDefault(Settings.EMPTY);
        MapperParsingException e = expectThrows(MapperParsingException.class, () -> mapper.parse(source(b -> {
            b.startObject("field");
            b.startArray("inner");
            for (int i = 1; i <= depth_limit; i++) {
                b.startArray();
            }
            for (int i = 1; i <= depth_limit; i++) {
                b.endArray();
            }
            b.endArray();
            b.endObject();
        })));
    }
```

Reply:
Hi @reta , thanks for pointing this out. Depth check should be done for nested arrays as well as it also as parsing is done recursively for nested levels. I have updated the PR to include this fix also.